operator/src/main/resources/scripts/introspectDomain.sh

```diff
@@ -74,6 +87,17 @@ for dir_var in DOMAIN_HOME JAVA_HOME WL_HOME MW_HOME ORACLE_HOME; do
   [ ! -d "${!dir_var}" ] && trace SEVERE "Missing ${dir_var} directory '${!dir_var}'." && exit 1
 done
 
+trace "DATA_HOME=${DATA_HOME}"
+
+#
+# DATA_HOME env variable exists implies override directory specified.  Attempt to create directory
+#
+if [ -n ${DATA_HOME} ] && [ ! -d ${DATA_HOME} ]; then
```

Comment:
I found the following description from some online docs for bash conditional exresssions but I changed it to use '! -z' as you suggested since it's probably more portable and compatible with different bash versions.

-z string
    True if the length of string is zero. 
string
-n string
    True if the length of string is non-zero.